packages/airnode-validator/src/ois/ois.ts

```diff
@@ -211,6 +212,92 @@ const ensureSingleParameterUsagePerEndpoint: ValidatorRefinement<OIS> = (ois, ct
   });
 };
 
+const ensureEndpointAndApiSpecificationParamsMatch: ValidatorRefinement<SchemaType<typeof baseOisSchema>> = (
+  ois,
+  ctx
+) => {
+  const { apiSpecifications, endpoints } = ois;
+
+  // Ensure every "apiSpecification.paths" parameter is defined in "endpoints"
+  forEach(apiSpecifications.paths, (pathData, rawPath) => {
+    forEach(pathData, (paramData, httpMethod) => {
+      const apiEndpoints = endpoints.filter(
+        ({ operation }) => operation.method === httpMethod && operation.path === rawPath
+      );
+      if (!apiEndpoints.length) return; // Missing endpoint for apiSpecification should only be a warning
+
+      apiEndpoints.forEach((endpoint) => {
+        paramData!.parameters.forEach((apiParam) => {
+          const allEndpointParams = [...endpoint.parameters, ...endpoint.fixedOperationParameters];
+          const endpointParam = allEndpointParams.find(
+            ({ operationParameter }) =>
+              operationParameter.in === apiParam.in && operationParameter.name === apiParam.name
+          );
+          if (!endpointParam) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: `Parameter "${apiParam.name}" not found in "fixedOperationParameters" nor "parameters"`,
```

Comment:
:detective: 
Nit: I believe this is a double-negative :smile: 
```suggestion
              message: `Parameter "${apiParam.name}" not found in "fixedOperationParameters" or "parameters"`,
```

Reply:
I stand corrected!  :smile:

Reply:
Actually @kolenic-martin and @amarthadan disliked this as well, so let me take your suggestion :D